File: problems/bubble-sort/bubble-sort.py
def bubblesort(l):
    """
    Runtime: O(n^2)
    """
    last = len(l)-1
    for i in range(last):
        for j in range(i+1, last):
            if l[i] > l[j]:
                l[i], l[j] = l[j], l[i]
    return l

print bubblesort([8,2,4,7,9,0,1,4,5,7,8,9])
print bubblesort([])
print bubblesort([1])
print bubblesort([1,3])

File: problems/bubble-sort/bubble-sort.go
package main

import "fmt"

func main() {
	fmt.Println("=== Bubblesort ===")
	fmt.Println(bubblesort([]int{8, 2, 4, 7, 9, 0, 1, 4, 5, 7, 8, 9}))
	fmt.Println(bubblesort([]int{}))
	fmt.Println(bubblesort([]int{1}))
	fmt.Println(bubblesort([]int{1, 3}))
}

func bubblesort(list []int) []int {
	last := len(list) - 1
	for i := 0; i < last; i++ {
		for j := i + 1; j < last; j++ {
			if list[i] > list[j] {
				list[i], list[j] = list[j], list[i]
			}
		}
	}
	return list
}
